packages/strapi-helper-plugin/lib/src/components/ImgPreview/index.js

```diff
@@ -0,0 +1,267 @@
+import ImgPreviewArrow from 'components/ImgPreviewArrow';
+import ImgPreviewRemoveIcon from 'components/ImgPreviewRemoveIcon';
+import ImgPreviewHint from 'components/ImgPreviewHint';
+
+import styles from './styles.scss';
+
+class ImgPreview extends React.Component {
+  state = {
+    imgURL: '',
+    isDraging: false,
+    isOver: false,
+    isImg: false,
+    position: 0,
+  };
+
+  componentDidMount() {
+    // We don't need the generateImgURL function here since the compo will
+    // always have an init value here
+    this.setState({
+        imgURL: get(this.props.files, ['0', 'url'], ''),
+        isImg: this.isPictureType(get(this.props.files, ['0', 'name'], '')),
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isUploading !== this.props.isUploading) {
+      const lastFile = nextProps.files.slice(-1)[0];
+      this.generateImgURL(lastFile);
+      this.updateFilePosition(nextProps.files.length - 1);
+    }
+  }
+
+  getFileType = (fileName) => fileName.split('.').slice(-1)[0];
+
+  /**
+   * [generateImgURL description]
+   * @param  {FileList} files
+   * @return {URL}
+   */
+  generateImgURL = (file) => {
+    if ( /\.(jpe?g|png|gif)$/i.test(file.name) ) {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        this.setState({
+          imgURL: reader.result,
+          isImg: true,
+        });
+      }
+
+      reader.readAsDataURL(file);
+    } else {
+      this.setState({ isImg: false, imgURL: file.name });
+    }
+  }
+
+  handleClick = (type) => {
+    const { position } = this.state;
+    const { files } = this.props;
+    let file;
+    let nextPosition;
+
+    switch (type) {
+      case 'right':
+        file = files[position + 1] || files[0];
+        nextPosition = files[position + 1] ? position + 1 : 0;
+        break;
+      case 'left':
+        file = files[position - 1] || files[files.length - 1];
+        nextPosition = files[position - 1] ? position - 1 : files.length - 1;
+        break;
+      default:
+        // Do nothing
+    }
+
+    if (!file.url) {
+      this.generateImgURL(file)
+    } else {
+      this.setState({ imgURL: file.url, isImg: this.isPictureType(file.url) });
+    }
+
+    this.updateFilePosition(nextPosition);
+  }
+
+  handleDragEnter = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ isDraging: true });
+  }
+
+  handleDragLeave = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ isDraging: false });
+  }
+
+  handleDragOver = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+  }
+
+  handleDrop = (e) => {
+    this.setState({ isDraging: false });
+    this.props.onDrop(e);
+  }
+
+  handleMouseEnter = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ isOver: true });
+  }
+
+  handleMouseLeave = (e) => this.setState({ isOver: false });
+
+  isPictureType = (fileName) => /\.(jpe?g|png|gif)$/i.test(fileName);
```

Comment:
It seems not the best way to check if a file is an image. You should base your logic on the type returned by the file. I can have a file named `test.jpg` which is a text file.

Reply:
I'll keep that in mind when I will implement this input in the content manager.
for the time being I feel that it will take me too much time to test another logic.

I'll add a note in the code